File: app/src/main/java/com/study/base/BaseActivity.java
package com.study.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * 基类activity
 */
public abstract class BaseActivity extends AppCompatActivity {


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        int layoutId = getLayoutId();
        setContentView(layoutId);
        findId();
    }

    protected abstract int getLayoutId();

    protected abstract void findId();
}

File: app/src/main/java/com/study/kt/BaseActivity.kt
package com.study.kt

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity: AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId())
        findViewById()
    }

    protected abstract fun layoutId():Int
    protected abstract fun findViewById()

}